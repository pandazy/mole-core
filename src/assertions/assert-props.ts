import { hasOwnProps } from '../object-methods';
import { assertValue } from './assert-value';

const hasProps = (fields: string[]) => {
	const propBelongsTo = hasOwnProps(fields);
	return (obj: any) => (obj != null ? propBelongsTo(obj) : false);
};

export function assertProps<TInput>(fields: string[]) {
	return (context: string) => (value: TInput) =>
		assertValue<TInput>({
			isValid: hasProps(fields),
			failureCause: `must have these properties: ${fields.join(', ')}`,
		})(context)(value);
}
