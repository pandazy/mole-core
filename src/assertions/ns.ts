import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`assertions:${context}`);
}
