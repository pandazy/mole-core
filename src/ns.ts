export function ns(context: string): string {
  return `Mole:${context}`;
}
