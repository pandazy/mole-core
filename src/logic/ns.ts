import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`logic:${context}`);
}
