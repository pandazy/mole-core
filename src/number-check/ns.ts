import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`numericCheck:${context}`);
}
