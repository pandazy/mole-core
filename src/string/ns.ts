import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`string:${context}`);
}
