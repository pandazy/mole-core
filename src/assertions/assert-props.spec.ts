import { assertProps } from './assert-props';
import { ns } from './ns';

describe(ns('assertProps'), () => {
  interface MockType {
    name: string;
    job?: string;
  }

  it('should assert fields existing in an input', () => {
    expect.assertions(3);

    const assertNameAge = assertProps<MockType>(['name', 'job'])(
      ns('assertProps'),
    );
    expect(() =>
      assertNameAge({ name: 'shinobu' }),
    ).toThrowErrorMatchingSnapshot();
    expect(() =>
      assertNameAge(undefined as any),
    ).toThrowErrorMatchingSnapshot();
    expect(() => assertNameAge(null as any)).toThrowErrorMatchingSnapshot();
    assertNameAge({ name: 'shinobu', job: 'jobless' });
  });
});
