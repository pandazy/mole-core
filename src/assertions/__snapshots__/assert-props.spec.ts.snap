// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mole:assertions:assertProps should assert fields existing in an input 1`] = `
"
[context]: Mole:assertions:assertProps
[failure cause]: must have these properties: name, job
[actual value]: {"name":"shinobu"}
[actual value type]: Object"
`;

exports[`Mole:assertions:assertProps should assert fields existing in an input 2`] = `
"
[context]: Mole:assertions:assertProps
[failure cause]: must have these properties: name, job
[actual value]: undefined
[actual value type]: Nil:(undefined)"
`;

exports[`Mole:assertions:assertProps should assert fields existing in an input 3`] = `
"
[context]: Mole:assertions:assertProps
[failure cause]: must have these properties: name, job
[actual value]: null
[actual value type]: Nil:(null)"
`;
